openshift-with-appstudio-test/e2e/jvm_build_service_test.go

```diff
@@ -340,26 +138,25 @@ func TestExampleRun(t *testing.T) {
 				return false, nil
 			}
 			if !pr.IsDone() {
-				prBytes, err := json.MarshalIndent(pr, "", "  ")
 				if err != nil {
 					ta.Logf(fmt.Sprintf("problem marshalling in progress pipelinerun to bytes: %s", err.Error()))
 					return false, nil
 				}
-				ta.Logf(fmt.Sprintf("in flight pipeline run: %s", string(prBytes)))
+				ta.Logf(fmt.Sprintf("in flight pipeline run: %s", pr.Name))
+				return false, nil
 			}
-			if !pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
+			if pr.IsDone() && !pr.GetStatusCondition().GetCondition(apis.ConditionSucceeded).IsTrue() {
 				prBytes, err := json.MarshalIndent(pr, "", "  ")
 				if err != nil {
 					ta.Logf(fmt.Sprintf("problem marshalling failed pipelinerun to bytes: %s", err.Error()))
 					return false, nil
 				}
-				ta.Logf(fmt.Sprintf("not yet successful pipeline run: %s", string(prBytes)))
-
+				return false, fmt.Errorf("pipeline run did not succeed: %s", string(prBytes))
 			}
 			return true, nil
 		})
 		if err != nil {
-			debugAndFailTest(ta, "timed out when waiting for the pipeline run to complete")
+			debugAndFailTest(ta, fmt.Sprintf("failure occured when waiting for the pipeline run to complete: %v", err))
```

Comment:
the error is going to be a "wait expried / timeout error" anyway

IMO this change does not provide any value

Reply:
Hmm I thought pollImmediate works in a way that it returns an error in cases
* of a timeout
* conditionFunc returns an error

so I was assuming the timeout is not always a cause of a failure and was thinking that in case the pipelinerun fails (not because of a timeout), it prints the error [from here](https://github.com/redhat-appstudio/jvm-build-service/blob/607e71dddc36a443b05b53e9bc5323bd1dd5a04e/openshift-with-appstudio-test/e2e/jvm_build_service_test.go#L154)

But I don't really mind to switch it back (especially if it doesn't work like that :)

Reply:
ah ... I missed that you are returning an error if the pipelinerun failed and is done

so nevermind leave it as is  @psturc